### DISCUSSION TITLE
Customized sorting function for queryContent

### DISCUSSION THREAD

**Discussion by xarthurx** [nuxt/content#2263] [Q&A] [answered]
## problem
I have some md-based posts, each with different time format. For instance:

```
- 2018-02-03T00:00:00+01:00
- July 2 2017
```

I'd like to sort  the posts based on the time, so I need to do something like:

```javascript
  posts.sort((a, b) => +new Date(b.date) - +new Date(a.date)
```

But the `queryContent`'s sort option doesn't allow customized sort function...

## alternatives tried
I can think of another approach of doing this, buy just collecting the post info in a `Array` and sort it.
But it seems the `await useAsyncData<Post>('xblog', () => queryContent('/blog').only([<some options>]).find()` returns a `Promise` object, and cannot simply be deconstructed into `Array`...

**Question: ** What should I do now...?

**Comment by nobkd** [comment, marked as answer]
I think this doesn't work with functions because they currently can't be stringified.
This is needed because the whole query is included in a request to an internal Nuxt API endpoint.
I can't find the corresponding issues/discussions... So I might be wrong...

---

Something like the following should work:
```vue
<script setup>
let { data: posts } = await useAsyncData('xblog', () => queryContent('/blog').only(['title', '_path', 'date']).find())
// do something with `posts`' ref
// e.g.
posts.value.sort((a, b) => new Date(b.date) - new Date(a.date))
// don't take my word here, I'm not sure when it comes to refs
</script>
<template>
  <pre>
  {{ posts }}
  </pre>
</template>
```

**Reply by xarthurx** [reply]
Yes, I just realised how `ref` is used in the return of `useAsyncData`, and how `Proxy` is returned instead of the raw objects.

The code seems to work. Thank you.

```typescript
console.log(postRaw.value.forEach( e=> console.log(e.date)))
postRaw.value.sort((a, b) => new Date(b.date) - new Date(a.date))

console.log("------")
console.log(postRaw.value.forEach( e=> console.log(e.date)))
```

Results:
```
------origin
ListPosts.vue:33 2018-02-02T23:00:00.000Z
ListPosts.vue:33 05Jul2023
ListPosts.vue:33 2023-04-06T14:54:25.000Z

------sorted
ListPosts.vue:37 05Jul2023
ListPosts.vue:37 2023-04-06T14:54:25.000Z
ListPosts.vue:37 2018-02-02T23:00:00.000Z

```
